hail/src/main/scala/is/hail/shuffler/LoggerCode.scala

```diff
@@ -0,0 +1,25 @@
+package is.hail.shuffler
+
+import is.hail.asm4s._
+import org.apache.log4j.Logger
+
+import scala.reflect.{ClassTag, classTag}
+
+object LoggerCode {
+  def getLogger[T: ClassTag](): Code[Logger] =
+    Code.invokeStatic1[Logger, String, Logger]("getLogger", classTag[T].runtimeClass.getName())
+}
+
+
+class LoggerCode(
+  code: Value[Logger]
+) extends Value[Logger] {
```

Comment:
The other pattern we normally use is an implicit conversion so you can write `log: Value[Logger]` or `log: Code[Logger]` in the code, and still say `log.info()`.  See RichCodeRegion for example.